test/integration/framework/plant_operations.go

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sirupsen/logrus"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// NewPlantTest creates a new shootGardenerTest object, given an already created shoot (created after parsing a shoot YAML)
+func NewPlantTest(kubeconfig string, plant *gardencorev1alpha1.Plant, shoot *v1beta1.Shoot, logger *logrus.Logger) (*PlantTest, error) {
```

Comment:
Why do we make our `Plant` integration test depending on a created `Shoot` object? It should be only depending on a kubeconfig, isn't it?